middleware.go

```diff
@@ -291,3 +291,24 @@ func EnableCORS(method string, path string) echo.MiddlewareFunc {
 		AllowMethods: []string{method},
 	})
 }
+
+type WEOSError struct {
```

Comment:
Make this a golang error similar to how a custom error was made in the weos lib. This will allow a developer to wrap an error within this weos controller error. So now the handlers will return the WeOS controller error you've created here. The way this helps is that if you get back a domain error from the module you can wrap that error in the WeOS error. In the customer handler you can look at the error type and say hey domain error that is status code 4XX. For errors that come back from the projection you can return a WeOS controller error WITH a status code defined depending on the situation

Reply:
Explanation of golang error handling features - https://blog.golang.org/go1.13-errors

Example of defining a custom error: https://github.com/wepala/weos/blob/0006285c4ee3e245664778d16a94efa7c53154e7/errors.go#L4

Reply:
thanks man noted will get to it